src/MongoDB.Bson/IO/JsonReader.cs

```diff
@@ -1036,17 +1045,22 @@ private void ParseBinDataExtendedJsonCanonical(out byte[] bytes, out BsonBinaryS
 
             if (base64String == null)
             {
-                var message = "$binary type wrapper is missing a 'base64' element.";
+                var message = "JSON reader expected $binary to contain a 'base64' element.";
                 throw new FormatException(message);
             }
             if (subTypeString == null)
             {
-                var message = "$binary type wrapper is missing a 'subType' element.";
+                var message = "JSON reader expected $binary to contain a 'subType' element.";
+                throw new FormatException(message);
+            }
+            if (!IsValidBinaryDataSubTypeString(subTypeString))
+            {
+                var message = string.Format("JSON reader expected subType to be a one or two digit hex string, but found '{0}'.", subTypeString);
                 throw new FormatException(message);
             }
 
             bytes = Convert.FromBase64String(base64String);
-            subType = (BsonBinarySubType)int.Parse(subTypeString);
+            subType = (BsonBinarySubType)Hex.ParseInt32(subTypeString);
```

Comment:
I see unit tests of IsValidBinaryDataSubTypeString, but can you add a test with a hex string within a $binary document? I may have missed it, so if so, ignore this.

Reply:
Added more tests to test hexadecimal parsing of subType strings.